prompb/remote.proto

```diff
@@ -23,10 +23,32 @@ message WriteRequest {
   repeated prometheus.TimeSeries timeseries = 1 [(gogoproto.nullable) = false];
 }
 
+// ReadRequest represents a remote read request.
 message ReadRequest {
   repeated Query queries = 1;
+
+  enum ResponseType {
+    // Server will stream a varint delimited ChunkedReadResponse message that contains XOR encoded chunks for a single series.
+    //
+    // Response headers:
+    // Content-Type: "application/x-streamed-protobuf; proto=prometheus.ChunkedReadResponse"
+    // Content-Encoding: ""
+    STREAMED_XOR_CHUNKS = 0;
+  }
+
+  // accepted_response_types allows negotiating the content type of the response.
+  //
+  // Response types are taken from the list in the FIFO order. If no response type in `accepted_response_types` is
+  // implemented by server, error is returned.
+  // For request that do not support `accepted_response_types` field the non streamed, raw samples response is used.
+  // In such case response headers are as follows:
+  //
+  // Content-Type: "application/x-protobuf"
```

Comment:
This should be the first item in the enum.

Reply:
See: https://github.com/prometheus/prometheus/pull/5701#discussion_r297801885

Reply:
Essentially the @tomwilkie idea was that SAMPLED is by default when the `accepted...` is empty, so no point in having separate enum item - and I can agree with that. It avoids having multiple options to control one thing.

What are you thoughts @brian-brazil why it should be in enum?

Reply:
What if I explicitly want to ask for it as a fallback?

Reply:
You can't.  It is the always the explicit fallback.  

If you want to ask for it explicitly, don't send any `accepted_response_types` - which is the same as getting a request for a client that doesn't know about this new field.

Reply:
That's not great in negotiation terms, maybe it's too inefficient for what I'm trying to do. I'd rather keep this explicit.

Reply:
Why is that not great?  Seems (a) pretty straight forward and (b) the correct embedding of what we want in what protos provide.

Reply:
We'd send a whole pile of data that the other end doesn't want. I can see this as a gotcha in the future, why is the server returning something in a format I didn't ask for?

Reply:
Yea I can see the failure in case of unsupported accepted formats only as the more explicit logic. If the client wants to be backward compatible it might need to explicitly state some `SAMPLES` enum at the end of `accepted_response_type` and this is fair. 

On the other hand @tomwilkie is right that there will be now two ways to ask for SAMPLES flow: either empty `accepted_response_type` or explicit SAMPLES in `accepted_response_type`. I feel like this is less evil though than the bit of the confusion as well as lack of way to disable SAMPLES response totally from client side. And I think it can be fair use case.

I am happy to revert to previous flow with `SAMPLES` and `STREAMED_XOR_CHUNKS` enum once we have consenus on this. Any other thoughts? @gouthamve ?